### PR TITLE
python: Update minor toolchain versions

### PR DESCRIPTION
**Pull request by phlax**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Other... Please describe:

Updates the python toolchain version for 3.10 -> 3.10.4

## What is the current behavior?

3.10.2 is installed

Issue Number: N/A

## What is the new behavior?

3.10.4 is installed

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

if you have 3.10.2 hardcoded as your version it would break

not sure if it is preferable to add 3.10.4 in addition rather than replacing existing version

### CODE REVIEW

**Comment by phlax** [comment]
i submitted this as i had been testing it out and it seemed useful 

i would be happy to update 3.8 and 3.9 also if that is helpful

File: python/versions.bzl [hunk 1]
```diff
@@ -58,14 +58,14 @@ TOOL_VERSIONS = {
         },
         "strip_prefix": "python",
     },
-    "3.10.2": {
-        "url": "20220227/cpython-{python_version}+20220227-{platform}-{build}.tar.gz",
+    "3.10.4": {
```
Comment (UebelAndre) [inline python/versions.bzl:61]: Would it be possible to leave `3.10.2` and instead add a new key for `3.10.4`?

**Review by f0rmiga — APPROVED** [review]

**Comment by f0rmiga** [comment]
@phlax I'd appreciate if you could add the new patch versions for the other minor ones.

**Comment by phlax** [comment]
minor versions added, but i just realised the msvc-shared pr has landed so i think this needs updating accordingly...

File: python/versions.bzl [hunk 2]
```diff
@@ -69,13 +91,24 @@ TOOL_VERSIONS = {
...
+            "x86_64-pc-windows-msvc": "bee24a3a5c83325215521d261d73a5207ab7060ef3481f76f69b4366744eb81d",
+            "x86_64-unknown-linux-gnu": "f6f871e53a7b1469c13f9bd7920ad98c4589e549acad8e5a1e14760fff3dd5c9",
+        },
+        "strip_prefix": "python",
+    },
 }
 
 # buildifier: disable=unsorted-dict-items
 MINOR_MAPPING = {
     "3.8": "3.8.12",
```
Comment (UebelAndre) [inline python/versions.bzl:109]: Would you be willing to also update `3.8` and `3.9` in a similar fashion to the `3.10` change?

**Pull request merged**